src/test/java/org/truffleruby/RubyTest.java

```diff
@@ -41,30 +43,31 @@
     protected void testWithAST(String text, Consumer<RubyRootNode> test) {
         final Source source = Source.newBuilder(text).name("test.rb").mimeType(RubyLanguage.MIME_TYPE).build();
 
-        testInEngine(() -> {
+        testInContext(() -> {
             final TranslatorDriver translator = new TranslatorDriver(RubyLanguage.getCurrentContext());
             final RubyRootNode rootNode = translator.parse(source, UTF8Encoding.INSTANCE, ParserContext.TOP_LEVEL, null, null, null, true, null);
             rootNode.adoptChildren();
             test.accept(rootNode);
         });
     }
 
-    protected void testInEngine(Runnable test) {
-        final PolyglotEngine engine = setupConfig(PolyglotEngine.newBuilder())
-                .globalSymbol("action", JavaInterop.asTruffleFunction(Runnable.class, test)).build();
-        try {
-            engine.eval(Source.newBuilder("Truffle::Interop.import('action').call").name("test.rb").mimeType(RubyLanguage.MIME_TYPE).build());
-        } finally {
-            engine.dispose();
+    protected void testInContext(Runnable test) {
+        final TruffleObject testTruffleObject = JavaInterop.asTruffleFunction(Runnable.class, test);
+
+        try (Context context = setupContext(Context.newBuilder()).build()) {
+            context.eval(org.graalvm.polyglot.Source.newBuilder(Launcher.LANGUAGE_ID, ":call.to_proc", "test.rb").build())
```

Comment:
`org.graalvm.polyglot.Source` should not need be fully qualified as it's `import`-ed.

Reply:
Ah, it's the different Source used above.